### DISCUSSION TITLE
Remove bloom filter requirement when using own node

### DISCUSSION THREAD

**Discussion by GordianLN** [bisq-network/bisq#5847] [Ideas]
I understand why and how bloom filters for SPV wallets are useful for privacy, I also reckon a node with bloom filters enabled might be a target to DDoS attacks.
Since when running a node on the same PC, to which Bisq connects automatically, or specifying a own node running elsewhere, you don't really need to hide things from yourself, but you sure would benefit from not running a bloom filters enabled node, it should be useful to have Bisq run without peerbloomfilters enabled, at least as an option.

**Comment by ripcurlx** [comment]
> I understand why and how bloom filters for SPV wallets are useful for privacy, I also reckon a node with bloom filters enabled might be a target to DDoS attacks. Since when running a node on the same PC, to which Bisq connects automatically, or specifying a own node running elsewhere, you don't really need to hide things from yourself, but you sure would benefit from not running a bloom filters enabled node, it should be useful to have Bisq run without peerbloomfilters enabled, at least as an option.

I don't remember exactly why we need this for bitcoinj, but I guess @chimp1984 might know why we can't get rid of it.

**Comment by chimp1984** [comment]
BitcoinJ supports theoretically a full node mode but that was never used by anybody in production as far I am aware. The SPV mode is requiring bloom filter as far I am aware. So I dont see an option to not use bloom filters for BitcoinJ.
